opendevin/core/utils/readers.py

```diff
@@ -0,0 +1,538 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+INSTALL
+-------
+pip install openai --upgrade
+pip install python-docx
+pip install markdown
+pip install PyPDF2
+pip install openpyxl
+pip install beautifulsoup4
+pip install pylatexenc
+pip install python-pptx
+pip install xlrd
+pip install base64
+pip install opencv-python
+"""
+
+import base64
+import json
+import subprocess
+import zipfile
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Any, Union
+
+import charset_normalizer
+import cv2
+import docx
+
+# import markdown
+import openpyxl
+import pandas as pd
+import PyPDF2
+import requests
+import yaml
+from bs4 import BeautifulSoup
+
+# from litellm import completion as litellm_completion
+# from opendevin.llm.llm import LLM
+from openai import OpenAI
+from pptx import Presentation
+from pylatexenc.latex2text import LatexNodes2Text
+
+from opendevin.core.logger import opendevin_logger as logger
+
+# TODO: Double-check where the interface for the stored API key is in OpenDevin.
+# TODO: Or ask someone familiar to change it with litellm.
+OPENAI_API_KEY = 'PUT_YOUR_OPEN_AI_API'
```

Comment:
You can take [here](https://github.com/OpenDevin/OpenDevin/blob/8750d3e68bb24fbcc954840a3f7ff2b101981ea1/opendevin/server/agent/agent.py#L108) as a reference.

Reply:
I agree. I simple use the config to get the API now.